### PR TITLE
add api status code description

### PR DESCRIPTION
**Pull request by vankichi**
Signed-off-by: vankichi <kyukawa315@gmail.com>

### Description:

I have created the gRPC status code overview document for the Vald cluster.
If there are more status codes that we are possible to get, let me know.

### Environment:

- Go Version: 1.18.2
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 1.14.5

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [x] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/api/status.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+- [Object Service](../api/object.md)
+- [Update Service](../api/update.md)
+- [Remove Service](../api/insert.md)
+- [Search Service](../api/search.md)
+
+The possible reasons are:
+
+- There is no index data in Vald Agent components or running the indexing process in the Vald Agent components.
+  When the Vald Agent component runs the indexing process (createIndex/saveIndex), any process won't run, and it will return with no result.
+- The request query vector or id is wrong when using the search/update/remove service.
```
[the remark this answers is not in the document]
Comment (vankichi) [reply docs/api/status.md:69]:
```suggestion
- The request query vector or ID is wrong when using the search/update/remove service.
```

File: docs/api/status.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+Services that return status are:
+
+- [Insert Service](../api/insert.md)
+- [Update Service](../api/update.md)
+
+You have to change the query vector with `skip_strict_exist_check` as `true` or set `skip_strict_exist_check` as `false` instead of changing the query vector.
+
+## RESOURCE_EXHAUSTED
+
+`RESOURCE_EXHAUSTED` means the gRPC message size is more extensive than the limit (default is 4MB).
```
[the remark this answers is not in the document]
Comment (vankichi) [reply docs/api/status.md:85]:
```suggestion
`RESOURCE_EXHAUSTED` means the gRPC message size is more extensive than the limit (default is 4 MB).
```

**Review by kmrmt — APPROVED** [review]

**Review by kevindiu — DISMISSED** [review]

**Review by hlts2 — APPROVED** [review]
LGTM

**Review by datelier — APPROVED** [review]

File: docs/api/status.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+Services that return status are:
+
+- [Insert Service](../api/insert.md)
+- [Update Service](../api/update.md)
+
+You have to change the query vector with `skip_strict_exist_check` as `true` or set `skip_strict_exist_check` as `false` instead of changing the query vector.
+
+## RESOURCE_EXHAUSTED
+
+`RESOURCE_EXHAUSTED` means the gRPC message size is more extensive than the limit (default is 4 MB).
```
Comment (kpango) [inline docs/api/status.md:85]: it's not only for gRPC payload resource means payload, network, server resource.

File: docs/api/status.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+Services that return status are all services.
+The most case in the Vald is that the query vector is too large.
+In other words, the vector dimension size in configuration is too large.
+
+## INTERNAL
+
+`INTERNAL` appears when some wrong happens in the Vald cluster.
+
+Services that return status are all services.
+If you get it, please verify the state of the Vald cluster.
```
Comment (kpango) [inline docs/api/status.md:96]: could you please add warning comment if Internal server error appears, it should be care and research about logs metrics.

**Review by kpango — CHANGES_REQUESTED** [review]

File: docs/api/status.md (hunk 3 above)
Comment (vankichi) [reply docs/api/status.md:85]: @kpango 
Does that mean in a Vald cluster? Or is it in a general sense?

Comment (kpango) [reply docs/api/status.md:85]: Its general sense

Comment (vankichi) [reply docs/api/status.md:85]: @kpango I have fixed doc.
could you please review it again?

Comment (kpango) [reply docs/api/status.md:85]: I think it wrong.
```
means the gRPC message size is more extensive than the limit (default is 4 MB).
```
it not only for gRPC payload

Comment (vankichi) [reply docs/api/status.md]: @kpango updated :)

File: docs/api/status.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+## RESOURCE_EXHAUSTED
+
+`RESOURCE_EXHAUSTED` means the some resources has been exhausted.
+
+It appears when:
+
+- There is out-of-memory in gRPC payload, or networks, or etc.
+- There are some server overload situations.
+- The Sent or received message is larger than configured limit (default is 4MB).
```
[the remark this answers is not in the document]
Comment (vankichi) [reply docs/api/status.md:91]:
```suggestion
- The Sent or received message is larger than the configured limit (default is 4 MB).
```

**Review by kpango — DISMISSED** [review]
LGTM

File: docs/api/status.md [hunk 6]
```diff
@@ -88,7 +88,7 @@ It appears when:
 
 - There is out-of-memory in gRPC payload, or networks, or etc.
 - There are some server overload situations.
-- The gRPC message size is more extensive than the limit (default is 4MB).
+- The Sent or received message is larger than configured limit (default is 4MB).
```
Comment (kevindiu) [inline docs/api/status.md]: I think the wording `Sent` should be `sent`

File: docs/api/status.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+## RESOURCE_EXHAUSTED
+
+`RESOURCE_EXHAUSTED` means the some resources has been exhausted.
+
+It appears when:
+
+- There is out-of-memory in gRPC payload, or networks, or etc.
+- There are some server overload situations.
+- The Sent or received message is larger than the configured limit (default is 4 MB).
```
Comment (vankichi) [inline docs/api/status.md:91]:
```suggestion
- The sent or received message is larger than the configured limit (default is 4 MB).
```

File: docs/api/status.md (hunk 6 above)
Comment (vankichi) [reply docs/api/status.md]: fixed

**Review by kevindiu — APPROVED** [review]

**Review by kmrmt — APPROVED** [review]

**Review by kpango — APPROVED** [review]

**Review by datelier — APPROVED** [review]

**Review by hlts2 — APPROVED** [review]

**Pull request merged**